src/Physics/BoostedDarkMatter/EventGen/DMEKinematicsGenerator.cxx

```diff
@@ -0,0 +1,256 @@
+//____________________________________________________________________________
+/*
+ Copyright (c) 2003-2019, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+ or see $GENIE/LICENSE
+
+ Author: Costas Andreopoulos <costas.andreopoulos \at stfc.ac.uk>
+         University of Liverpool & STFC Rutherford Appleton Lab 
+
+ For the class documentation see the corresponding header file.
+
+ Important revisions after version 2.0.0 :
+ @ Feb 09, 2009 - CA
+   Moved into the NuE package from its previous location (EVGModules package)
+ @ Feb 06, 2013 - CA
+   When the value of the differential cross-section for the selected kinematics
+   is set to the event, set the corresponding KinePhaseSpace_t value too.
+
+*/
+//____________________________________________________________________________
+
```

Comment:
Some large amounts of duplication between the neutrino and DM event generation code - like in this case. While this doesn't prevent accepting the pull request and merge, could we look at reusing (rather than re-implementing) in a future iteration? The code that handles the event generation (for example for generation of event kinematics) is mostly "mechanics", and the physics is  (or should be) contained within the associated cross-section model. This will make it much easier to continue maintaining the BDM module in future releases, and to make sure that BDM benefits from improvements installed in the generic generation modules).